### PR TITLE
More strong typing

### LINKED ISSUE #1813: Cannot delete groupfolder with occ command
Hi,
I'm using NC 23.0.0.10 in a docker container and have some group folders:

```www-data@52365b656dc1:~/html$ php occ groupfolders:list

+-----------+------------------+-----------------------------------+-----------+----------+----------------------+
| Folder Id | Name             | Groups                            | Quota     | Size     | Advanced Permissions |
+-----------+------------------+-----------------------------------+-----------+----------+----------------------+
| 2         | Alle Bilder      | <groupname>: read, write, share, delete | Unlimited | 19.5 GB  | Enabled              |    
| 3         | Musik            | <groupname>: read, write, share, delete | Unlimited | 520 B    | Enabled              |      
| 4         | Videos           | <groupname>: read, write, share, delete | Unlimited | 196.8 GB | Enabled              |       
| 6         | Bilder sonst     | <groupname>: read, write, share, delete | Unlimited | 6.1 GB   | Enabled              |          
| 7         | Bilder Fotoalben | <groupname>: read, write, share, delete | Unlimited | 2.8 GB   | Enabled              |        
+-----------+------------------+-----------------------------------+-----------+----------+----------------------+
```

### Steps to reproduce
``php occ groupfolders:delete 3``
Also tried via website without success.

### Expected behaviour
Group folder with id 3 shall be deleted

### Actual behaviour
Exception:
```
An unhandled exception has been thrown:
TypeError: OCA\GroupFolders\Folder\FolderManager::getFolder(): Argument nextcloud/server#1 ($id) must be of type int, string given, called in /var/www/html/apps/groupfolders/lib/Command/Delete.php on line 57 and defined in /var/www/html/apps/groupfolders/lib/Folder/FolderManager.php:213
Stack trace:
#0 /var/www/html/apps/groupfolders/lib/Command/Delete.php(57): OCA\GroupFolders\Folder\FolderManager->getFolder('3', 1)
nextcloud/server#1 /var/www/html/3rdparty/symfony/console/Command/Command.php(255): OCA\GroupFolders\Command\Delete->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#2 /var/www/html/core/Command/Base.php(168): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#3 /var/www/html/3rdparty/symfony/console/Application.php(1009): OC\Core\Command\Base->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#4 /var/www/html/3rdparty/symfony/console/Application.php(273): Symfony\Component\Console\Application->doRunCommand(Object(OCA\GroupFolders\Command\Delete), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#5 /var/www/html/3rdparty/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#6 /var/www/html/lib/private/Console/Application.php(211): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
nextcloud/server#7 /var/www/html/console.php(99): OC\Console\Application->run()
nextcloud/server#8 /var/www/html/occ(11): require_once('/var/www/html/c...')
```

### Server configuration

Operating system:  Debian 5.10.70-1
Nextcloud docker image "nextcloud:latest" with ID e078e5cc644a
Nextcloud version: 23.0.0.10

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "**REMOVED**"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "23.0.0.10",
        "overwrite.cli.url": "**REMOVED**",
        "overwriteprotocol": "https",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "default_phone_region": "DE",
        "default_language": "de",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "twofactor_yubikey",
            "mindmap_app",
            "groupfolders",
            "files_frommail",
            "previewgenerator"
        ],
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "log_rotate_size": 52428800,
        "log_type": "syslog",
        "logfile": "",
        "loglevel": 3,
        "logtimezone": "Europe\/Berlin",
        "maintenance": false,
        "has_rebuilt_cache": true
    }
}
```

### PR DESCRIPTION
**Pull request by CarlSchwan**
Fix #1813

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

**Comment by oresch** [comment]
As there is no vi or something like that in the docker image (or I didn't find it and I'm not able to install vi), I currently cannot test that fix. :-(

**Review by icewind1991 — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
+		"php-parallel-lint/php-parallel-lint": "^1.2"
+	},
+	"scripts": {
+		"lint": "find . -name \\*.php -not -path './vendor/*' -print0 | xargs -0 -n1 php -l",
+		"cs:check": "php-cs-fixer fix --dry-run --diff",
+		"cs:fix": "php-cs-fixer fix",
+		"psalm": "psalm --threads=1",
+		"psalm:update-baseline": "psalm --threads=1 --update-baseline --set-baseline=tests/psalm-baseline.xml",
+		"psalm:clear": "psalm --clear-cache && psalm --clear-global-cache",
+		"psalm:fix": "psalm --alter --issues=InvalidReturnType,InvalidNullableReturnType,MissingParamType,InvalidFalsableReturnType"
```
Comment (juliushaertl) [inline composer.json:21]:
```suggestion
		"psalm": "psalm.phar --threads=1",
		"psalm:update-baseline": "psalm.phar --threads=1 --update-baseline --set-baseline=tests/psalm-baseline.xml",
		"psalm:clear": "psalm.phar --clear-cache && psalm --clear-global-cache",
		"psalm:fix": "psalm.phar --alter --issues=InvalidReturnType,InvalidNullableReturnType,MissingParamType,InvalidFalsableReturnType"
```

**Review by juliushaertl — APPROVED** [review]

**Comment by guystreeter** [comment]
Are you sure this is correct? Using `(int)$folderId = ...`  did not work for me, but `$folderId = (int)...` did.

**Pull request merged**